tests/src/main/java/org/hibernate/beanvalidation/tck/tests/AbstractBootstrapFailureTCKTest.java

```diff
@@ -0,0 +1,113 @@
+ * <p>
+ * In some cases, the failure can either happen at deployment or at bootstrap. Thus, with this class, when the
+ * deployment is failing, we catch the deployment error if it matches a certain type of exception, and we consider it at
+ * the exception that will be thrown by the tests.
+ * <p>
+ * In this case, the content of the tests is not executed so you need to be very careful to use this class only when
+ * this behavior is required.
+ *
+ * @author Guillaume Smet
+ */
+public abstract class AbstractBootstrapFailureTCKTest extends AbstractTCKTest {
+
+	private Exception acceptedDeploymentException;
+
+	protected abstract Class<? extends Exception> acceptedDeploymentExceptionType();
+
+	@Override
+	@BeforeClass(groups = "arquillian", inheritGroups = true)
+	public void arquillianBeforeClass() throws Exception {
+		try {
+			super.arquillianBeforeClass();
+		}
+		catch (Exception e) {
+			// if the deployment throws an exception and if it's of the right type, we don't throw the exception here
+			// and we will "virtually" throw it instead of executing the tests
+			acceptedDeploymentException = extractAcceptedDeploymentException( acceptedDeploymentExceptionType(), e );
+			if ( acceptedDeploymentException == null ) {
+				throw e;
+			}
+		}
+	}
+
+	@Override
+	@BeforeMethod(groups = "arquillian", inheritGroups = true)
+	public void arquillianBeforeTest(Method testMethod) throws Exception {
+		// Don't do anything if the deployment failed
+		if ( acceptedDeploymentException != null ) {
+			return;
+		}
+
+		super.arquillianBeforeTest( testMethod );
+	}
+
+	@Override
+	public void run(IHookCallBack callback, ITestResult testResult) {
+		if ( acceptedDeploymentException != null ) {
+			// We consider that the test failed with the exception caught during the deployment
+			testResult.setStatus( ITestResult.FAILURE );
+			testResult.setThrowable( acceptedDeploymentException );
+
+			return;
+		}
+
+		super.run( callback, testResult );
+	}
+
+	private static Exception extractAcceptedDeploymentException(Class<? extends Exception> acceptedDeploymentExceptionType, Throwable exception) {
+		if ( acceptedDeploymentExceptionType.isAssignableFrom( exception.getClass() ) ) {
+			return (Exception) exception;
+		}
+
+		if ( exception instanceof InvocationTargetException ) {
+			Exception acceptedDeploymentException = extractAcceptedDeploymentException( acceptedDeploymentExceptionType,
+					( (InvocationTargetException) exception ).getTargetException() );
+			if ( acceptedDeploymentException != null ) {
+				return acceptedDeploymentException;
+			}
+		}
+
+		if ( exception.getCause() != null ) {
+			Exception acceptedDeploymentException = extractAcceptedDeploymentException( acceptedDeploymentExceptionType, exception.getCause() );
+			if ( acceptedDeploymentException != null ) {
+				return acceptedDeploymentException;
+			}
+		}
+
+		// in the case of Weld, the wrapped exceptions are simply stored in the message, so we check for the exception
```

Comment:
No big fan of this. Not sure thought what could be done alternatively... .

Reply:
As you can imagine, I'm no fan either.

Unfortunately, we don't have much of a choice as the exception is lost in CDI. See https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/exceptions/DeploymentException.java#L52 . We don't have a way to get back the cause.

I asked @aguibert if he was in the case where there was only one exception in the list because, if so, we could see with the Weld people if they would accept to create a static method taking the list instead of a constructor and special case the "List with only one cause" case. @aguibert still interested in the answer :).

We would still have to wait for a Weld release though so we can hope for better in the future, but for now, we will have to accept that.

Reply:
Yes, all the cases I saw were just one exception in message.  
However, I don't think this is worth opening an issue with the Weld folks because, in practice, no java code (aside from test frameworks) are going to care if the exception is a proper chained exception or simply in the exception message.